### DISCUSSION TITLE
[Question]: Use selfhosted AppWrite in immich stack ?

### DISCUSSION THREAD

**Discussion by Nonobis** [immich-app/immich#216] [General]
### Feature detail

Could using AppWrite (self hosted Firebase equivalent) could be a good idea for Immich ? 

https://github.com/appwrite/appwrite#client

AppWrite is multi-languages, and fully self hosted. There is a lot of service who could be useful for immich i think ...

Services
[Account](https://appwrite.io/docs/client/account) - Manage current user authentication and account. Track and manage the user sessions, devices, sign-in methods, and security logs.
[Users](https://appwrite.io/docs/server/users) - Manage and list all project users when in admin mode.
[Teams](https://appwrite.io/docs/client/teams) - Manage and group users in teams. Manage memberships, invites, and user roles within a team.
[Database](https://appwrite.io/docs/client/database) - Manage database collections and documents. Read, create, update, and delete documents and filter lists of document collections using advanced filters.
[Storage](https://appwrite.io/docs/client/storage) - Manage storage files. Read, create, delete, and preview files. Manipulate the preview of your files to fit your app perfectly. All files are scanned by ClamAV and stored in a secure and encrypted way.
[Functions](https://appwrite.io/docs/server/functions) - Customize your Appwrite server by executing your custom code in a secure, isolated environment. You can trigger your code on any Appwrite system event, manually or using a CRON schedule.
[Locale](https://appwrite.io/docs/client/locale) - Track your user's location, and manage your app locale-based data.
[Avatars](https://appwrite.io/docs/client/avatars) - Manage your users' avatars, countries' flags, browser icons, credit card symbols, and generate QR codes.

### Platform

Server

**Comment by alextran1502** [comment]
This would require the entire backend to be migrated and Immich will be dependent on a different solution. This is not suitable for Immich and the current development plan
